spec/lib/exchange_rate_adapters/average_rate_adapter_spec.rb

```diff
@@ -0,0 +1,17 @@
+require 'spec_helper'
+
+RSpec.describe Straight::ExchangeRate::AverageRateAdapter do
+
+  before(:all) do
+    @average_rates_adapter =  Straight::ExchangeRate::AverageRateAdapter.new(
+                                Straight::ExchangeRate::BitstampAdapter.new, 
+                                Straight::ExchangeRate::BitpayAdapter.new)
+  end
```

Comment:
This should be changed in the constructor. If I don't want to pass adapter instances, then the constructor should create them for me. So, I must have the freedom of passing both instances and classes. For example:

    Straight::ExchangeRate::AverageRateAdapter.new(
      Straight::ExchangeRate::BitstampAdapter.new,
      Straight::ExchangeRate::BitpayAdapter
    )